src/system.rs

```diff
@@ -406,14 +482,19 @@ impl TypeStore {
             .or_else(|| self.imported_type_ids_by_name.get(name))
     }
 
-    /// Provides the [TypeId] associated with given [CoreISLTypeDefinition] if it exists in the [TypeStore]  
+    /// Provides the [TypeId] associated with given type name if it exists in the [TypeStore]  
     /// Otherwise returns None
-    pub fn get_core_type_id(&self, core_isl_type: &CoreISLTypeDefinition) -> Option<&TypeId> {
-        let core_type_name = match core_isl_type {
-            CoreISLTypeDefinition::IonType(ion_type) => format!("{}", ion_type),
-            CoreISLTypeDefinition::Other(other_type) => other_type.name().to_owned().unwrap(),
+    pub fn get_builtin_type_id(&self, type_name: &str) -> Option<TypeId> {
+        let type_name = match type_name {
+            "int" => "integer",
+            "bool" => "boolean",
+            "$int" => "$integer",
+            "$bool" => "$boolean",
```

Comment:
This looks like it might be a band-aid. Why does the type store use names that aren't the ones from the Ion spec?

Reply:
Because I am using `IonType` which has different naming than [Ion Schema Spec](https://amzn.github.io/ion-schema/docs/spec.html#type-system). I can make the type_store use names according to Ion Schema spec but then when validating I would have to convert all those strings to `IonType` and then validate. So either way will have to use a band-aid.

Reply:
As a TODO: let's make a dedicated function to get the Ion Schema name from a given `IonType` rather than doing mappings on the spot. Then we can document that function so readers understand why the names are different.